perfdash/metrics-downloader.go

```diff
@@ -279,6 +285,109 @@ func (b *GCSMetricsBucket) GetFile(job string, buildNumber int, path string) ([]
 	return data, nil
 }
 
+// S3MetricsBucket that creates a AWS S3 client to fetch data.
+type S3MetricsBucket struct {
+	client  *s3.S3
+	bucket  *string
+	logPath string
+}
+
+// NewS3MetricsBucket creates a new S3MetricsBucket.
+func NewS3MetricsBucket(bucket, path, region string) (MetricsBucket, error) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region)},
```

Comment:
nit: Can you turn on verbose credential chain debugging?
```suggestion
		Region:                        aws.String(region),
	        CredentialsChainVerboseErrors: aws.Bool(true)},
```